mautic/mautic
# New INTEGRATION_BEFORE_COMPANY_FIELD_CHANGES event

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [x] 
| Related user documentation PR URL      | /
| Related developer documentation PR URL | /
| Issue(s) addressed                     | /


#### Description:

In the integration we built on top of the IntegrationsBundle we needed a way how an integration can decide what field changes will be reported for sync before they are saved. This PR introduces new INTEGRATION_BEFORE_CONTACT_FIELD_CHANGES & INTEGRATION_BEFORE_COMPANY_FIELD_CHANGES that allows all integrations to do so.

#### Steps to test this PR:

There are no steps to test as an integration (plugin) that subscribes for this new event is needed to test it manually. However, this PR has all changes covered by tests to ensure no bug is introduced. Also, we run this code in production.
